crates/bevy_ecs/src/archetype.rs

```diff
@@ -536,6 +573,42 @@ impl Archetype {
     pub(crate) fn clear_entities(&mut self) {
         self.entities.clear();
     }
+
```

Comment:
Would you consider something like `has_hook(&self, hook_flag: ArchetypeFlags)` vs the repetition?

Reply:
Yeah for sure, this is overly verbose, this PR needs some cleanup generally before it will come out of draft. I'm just waiting on hooks to be merged.